crates/core/tedge_agent/src/mqtt_operation_converter/error.rs

```diff
@@ -0,0 +1,17 @@
+use tedge_actors::RuntimeError;
+
+#[derive(Debug, thiserror::Error)]
+#[allow(clippy::enum_variant_names)]
+pub enum MqttRequestConverterError {
+    #[error(transparent)]
+    FromSerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    FromChannelError(#[from] tedge_actors::ChannelError),
```

Comment:
It's a bit surprising to see channel errors included as a sub-case of converter error. It seems that two unrelated aspects are bound: message translation and interaction with peers.

Reply:
The file has been renamed - but the comment still applies.

Reply:
But converter sends requests, and `send()` can fail. The error is used in these kind of context.

```rust
            "tedge/commands/req/software/list" => {
                let request = SoftwareListRequest::from_slice(message.payload_bytes())?;
                self.software_sender
                    .send(SoftwareRequest::SoftwareListRequest(request))
                    .await?;
            }
```

Reply:
Yes. But a `ChannelError` is already a kind of `RuntimeError`. Here, you are wrapping a `ChannelError` into a `TedgeOperationConverterError` and a `TedgeOperationConverterError` into a `RuntimeError`.